toolkit/tools/internal/testutils/testrpms/download-test-utils.sh

```diff
@@ -45,16 +45,52 @@ esac
 
 set -x
 
-# call the script to create the tools file if requested
+# Declarative configuration
+TESTDATA_DIR="$SCRIPT_DIR/../../../pkg/imagecreatorlib/testdata"
+DISTRO="azurelinux"
+CONFIG_FILE="minimal-os.yaml"
+
+# Initialize package list
+PACKAGE_LIST=""
+
+# Handle tools file creation and package extraction for imagecreator testing
 if [ "$IMAGE_CREATOR" = "true" ]; then
   echo "Creating tools file: $TOOLS_FILE"
   $SCRIPT_DIR/create-tools-file.sh "$CONTAINER_IMAGE" "$TOOLS_FILE"
   echo "Tools file created successfully."
+
+  # Check for python3 availability
+  if ! command -v python3 >/dev/null 2>&1; then
+    echo "Error: python3 is required but not found in PATH"
+    echo "Please install python3 to extract package lists from config files"
+    exit 1
+  fi
+
+  # Extract package list from config file
+  CONFIG_PATH="$TESTDATA_DIR/$CONFIG_FILE"
+  if [[ -f "$CONFIG_PATH" ]]; then
```

Comment:
Reverse this `if` statement. Then, remove the `else` block because `exit 1` makes it redundant.